system-addon/content-src/components/LinkMenu/LinkMenu.jsx

```diff
@@ -1,52 +1,98 @@
 const React = require("react");
 const {injectIntl} = require("react-intl");
 const ContextMenu = require("content-src/components/ContextMenu/ContextMenu");
+const {actionTypes, actionCreators} = require("common/Actions.jsm");
 
 class LinkMenu extends React.Component {
-  constructor(props) {
-    super(props);
-    this.getOptions = this.getOptions.bind(this);
+  getBookmarkStatus(site) {
+    return (site.bookmarkGuid ? {
+      id: "menu_action_remove_bookmark",
+      icon: "bookmark-remove",
+      action: actionTypes.DELETE_BOOKMARK_BY_ID,
+      data: site.bookmarkGuid
+    } : {
+      id: "menu_action_bookmark",
+      icon: "bookmark",
+      action: actionTypes.BOOKMARK_URL,
+      data: site.url
+    });
+  }
+  getDefaultContextMenu(dispatch, site) {
+    return [{
+      id: "menu_action_open_new_window",
+      icon: "new-window",
+      onClick: () => {
+        dispatch(actionCreators.SendToMain({type: actionTypes.OPEN_NEW_WINDOW, data: {url: site.url}}));
```

Comment:
It looks like all of the options have an onClick of `dispatch`ing `SendToMain`, so it each of the options could probably just have an `action` and `data` that gets hooked up for all options.